scripts/check-sidebar-lazy-layout.py

```
@@ -1,741 +1,388 @@
+
+    for filename, source in sources_by_name.items():
+        clean = neutralize_swift(source)
+        for callback in LAYOUT_CALLBACKS:
+            body = extract_function_body(clean, callback)
+            if body is None:
+                continue
+            for pattern in LAYOUT_MUTATION_PATTERNS:
+                if pattern.search(body):
+                    violations.append(
+                        f"{filename}.{callback} mutates/reconfigures the table from a layout callback"
                     )
```

Ruff (0.15.21)

[warning] Use `list.extend` to create a transformed list

(PERF401)